packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts

```diff
@@ -258,6 +259,9 @@ export class CartCheckoutPage {
 	 * @param {string} cardHolderName Name of the card holder associated with the payment method.
 	 */
 	async selectSavedCard( cardHolderName: string ): Promise< void > {
+		if ( await this.page.isVisible( selectors.editPaymentStep ) ) {
```

Comment:
Could you add a comment explaining in what cases that scenario happens and why the condition is needed? At first glance, this might smell like something that might need to be split into another helper function for another scenario. Still, a comment will probably help clarify if this is a temporary or permanent solution. Thanks!

Reply:
👍 I added a comment. It seems like a good permanent solution to me since most tests don't care if the payment method step is auto-completed or not; if you call `selectSavedCard` you probably want to select a non-default card anyway. But I'm open to suggestions.

Reply:
Sounds reasonable. Thanks!